src/frontend/src/binder/expr/column.rs

```diff
@@ -36,19 +36,31 @@ impl Binder {
             }
         };
 
-        let result = self
+        match self
             .context
-            .get_column_binding_index(&table_name, &column_name);
-        match result {
-            Ok(index) => {
-                let column = &self.context.columns[index];
-                return Ok(InputRef::new(column.index, column.field.data_type.clone()).into());
+            .get_column_binding_index(&table_name, &column_name)
+        {
+            Ok(mut indices) => {
+                match indices.len() {
+                    0 => unreachable!(),
+                    1 => {
+                        let index = indices[0];
+                        let column = &self.context.columns[index];
+                        return Ok(InputRef::new(column.index, column.field.data_type.clone()).into());
+                    }
+                    _ => {
+                        indices.sort(); // make sure we have a consistent result
+                        let inputs = indices.iter().map(|index| {
+                            let column = &self.context.columns[*index];
+                            InputRef::new(column.index, column.field.data_type.clone()).into()
+                        }).collect::<Vec<_>>();
+                        return Ok(FunctionCall::new(ExprType::Coalesce, inputs)?.into())
+                    }
+                }
             }
             Err(e) => {
-                // If the error message is not that the column is not found, throw the error
-                if let ErrorCode::ItemNotFound(_) = e.inner() {
-                } else {
-                    return Err(e);
+                if let ErrorCode::InternalError(s) = e.inner() && s.starts_with("Ambiguous column name:") {
+                    return Err(e)
```

Comment:
Why this needs to be changed?

Reply:
Oh yes old one is actually better behaviour, I'll change it back.